source/dub/generators/build.d

```diff
@@ -371,12 +371,30 @@ class BuildGenerator : ProjectGenerator {
 
 	private void copyTargetFile(NativePath build_path, BuildSettings buildsettings, GeneratorSettings settings)
 	{
-		auto filename = settings.compiler.getTargetFileName(buildsettings, settings.platform);
-		auto src = build_path ~ filename;
-		logDiagnostic("Copying target from %s to %s", src.toNativeString(), buildsettings.targetPath);
 		if (!existsFile(NativePath(buildsettings.targetPath)))
 			mkdirRecurse(buildsettings.targetPath);
-		hardLinkFile(src, NativePath(buildsettings.targetPath) ~ filename, true);
+
+		string[] filenames = [
+			settings.compiler.getTargetFileName(buildsettings, settings.platform)
+		];
+
+		// Windows: add .pdb if found
+		const tt = buildsettings.targetType;
+		if ((tt == TargetType.executable || tt == TargetType.dynamicLibrary) &&
+		    settings.platform.platform.canFind("windows"))
+		{
+			import std.path : setExtension;
+			const pdbFilename = filenames[0].setExtension(".pdb");
+			if (existsFile(build_path ~ pdbFilename))
```

Comment:
Even with `-g` or `-gc`, things such as `/PDB:dir\binary.pdb` as lflags, a presumably common current workaround, prevent the generation of the default pdb, as well as stuff like `-gdwarf` with LDC. So check if the file exists and only copy in that case.